superset-frontend/packages/superset-core/src/api/sqlLab.ts

```diff
@@ -111,6 +111,171 @@ export interface Tab {
   panels: Panel[];
 }
 
+/**
+ * Generic data types, see enum of the same name in superset/utils/core.py.
+ */
+enum GenericDataType {
+  Numeric = 0,
+  String = 1,
+  Temporal = 2,
+  Boolean = 3,
+}
+
+/**
+ * Column metadata returned in query results.
+ */
+export type QueryColumn = {
+  /**
+   * Label of the column
+   */
+  name?: string;
+
+  /**
+   * Column name defined
+   */
+  column_name: string;
```

Comment:
If we had something like Thrift configured to generate both Typescript and Python classes, we would configure it so that Typescript classes would follow the camelCase convention and Python classes would follow the snake_case convention. Something like:

```
// Define in Thrift with consistent naming
struct QueryRequest {
  1: i32 query_limit
  2: string template_params
}

// Configure generators:
// - TypeScript: --gen js:ts,snake_case_to_camel_case
// - Python: --gen py (keeps snake_case naturally)
```

Given that we're defining Typescript APIs, let's make them all camelCase and handle the conversion manually for now. We can introduce an automatic conversion layer later.